go-controller/pkg/ovn/base_network_controller_policy.go

```diff
@@ -1490,6 +1494,39 @@ func (bnc *BaseNetworkController) peerNamespaceUpdate(np *networkPolicy, gp *gre
 	return err
 }
 
+// requeuePeerNamespace enqueues the namespace into network policy peer namespace
+// retry framework object(s) which need to be retried immediately with add event.
+func (bnc *BaseNetworkController) requeuePeerNamespace(namespace *corev1.Namespace) error {
+	if namespace == nil {
+		return nil
+	}
+	npKeys := bnc.networkPolicies.GetKeys()
+	var errors []error
+	for _, npKey := range npKeys {
+		err := bnc.networkPolicies.DoWithLock(npKey, func(npKey string) error {
```

Comment:
Just an observation here. I wonder why we have both `DoWithLock` and `np.Mutex`. Seems like we don't need both of them in general.

Reply:
AFAIU `DoWithLock` is needed to synchronize accessing map entry whereas `np.Mutex` is used to synchrnonize accessing network policy object itself. this is the pattern used across np code, so not changing that in this PR.

Reply:
I don't think that's it. `GetKeys` and `Load` already constitute a synchronized access to the map. `DoWithLock` should cover up for `np.Mutex` if used broadly and correctly, and additionally prevent for a key being removed from the map while being used. So I guess I am missing the point of `np.Mutex`.
However, I didn't ask to do anything, hence just the observation. Agree to continue the current usage pattern for now.